src/opf/powerflow.py

```diff
@@ -373,5 +409,7 @@ def powerflow(
     #     Sbus_branch[i] += torch.sum(Sf[params.fr_bus == i])
     #     Sbus_branch[i] += torch.sum(St[params.to_bus == i])
     Sbus = Sbus_branch + Sbus_sh
-    S = Sg - Sd
+    Sg_unfiltered = torch.zeros_like(Sd)
+    Sg_unfiltered[params.gen_bus_ids.int()] = Sg
```

Comment:
You can use `torch.index_add` or `torch.index_add_`. The latter is an inplace version.
```
Sg_bus = torch.index_add(dim=0, index=powerflow_paramters.gen_bus_ids, source=Sg_gen)
```
Your current implementation will only work if there is one generator per bus (which is typically the case).